ecfr-parser/parseXML/parseXML.go

```diff
@@ -0,0 +1,322 @@
+	case "DIV8":
+		child := &Section{}
+		if err := d.DecodeElement(child, &start); err != nil {
+			return err
+		}
+		*c = append(*c, child)
+	default:
+		//return fmt.Errorf("Unknown XML type in Subpart: %+v", start)
+		log.Printf("[WARNING] Unknown XML type in Subject Group: %+v\n", start)
+		d.Skip()
+	}
+	return nil
+}
+
+type Section struct {
+	Type     string          `xml:"TYPE,attr" json:"node_type"`
+	Citation SectionCitation `xml:"N,attr" json:"label"`
+	Header   string          `xml:"HEAD" json:"title"`
+	Children SectionChildren `xml:",any" json:"children"`
+}
+
+func (s *Section) PostProcess() error {
+	var prev *Paragraph
+	for _, child := range s.Children {
+		c, ok := child.(*Paragraph)
+		if ok {
+			var err error
+			c.Citation, err = generateParagraphCitation(c, prev)
+			if err != nil && err != ErrNoParents {
+				e, l := c.Marker()
+				log.Println("[ERROR]", err, prev, c, e, l, s.Citation)
+			} else {
+				prev = c
+			}
+
+		}
+	}
+	for _, child := range s.Children {
+		c, ok := child.(*Paragraph)
+		if ok {
+			if len(c.Citation) > 0 {
+				cit := append([]string{}, s.Citation...)
+				c.Citation = append(cit, c.Citation...)
+			} else {
+				cit := append([]string{}, s.Citation...)
+				c.Citation = append(cit, c.Citation...)
+				c.Citation = append(c.Citation, fmt.Sprintf("%x", md5.Sum([]byte(c.Content))))
+			}
+		}
+	}
+
+	for _, child := range s.Children {
+		c, ok := child.(PostProcesser)
+		if ok {
+			if err := c.PostProcess(); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+type SectionChildren []interface{}
+
+func (c *SectionChildren) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	switch start.Name.Local {
+	case "P":
+		child := &Paragraph{Type: "Paragraph"}
+		if err := d.DecodeElement(child, &start); err != nil {
+			return err
+		}
+		*c = append(*c, child)
+	case "FP":
+		child := &FlushParagraph{Type: "FlushParagraph"}
+		if err := d.DecodeElement(child, &start); err != nil {
+			return err
+		}
+		*c = append(*c, child)
+	case "img":
+		child := &Image{Type: "Image"}
+		if err := d.DecodeElement(child, &start); err != nil {
+			return err
+		}
+		*c = append(*c, child)
+	case "EXTRACT":
+		child := &Extract{Type: "Extract"}
+		if err := d.DecodeElement(child, &start); err != nil {
+			return err
+		}
+		*c = append(*c, child)
+	case "CITA":
+		child := &Citation{Type: "Citation"}
+		if err := d.DecodeElement(child, &start); err != nil {
+			return err
+		}
+		*c = append(*c, child)
+	case "SECAUTH":
+		child := &SectionAuthority{Type: "SectionAuthority"}
+		if err := d.DecodeElement(child, &start); err != nil {
+			return err
+		}
+		*c = append(*c, child)
+	default:
+		//return fmt.Errorf("Unknown XML type in Section: %+v", start)
+		log.Printf("[WARNING] Unknown XML type in Section: %+v\n", start)
+		d.Skip()
+	}
+
+	return nil
+}
+
+type SectionCitation []string
+
+func (sl *SectionCitation) UnmarshalText(data []byte) error {
+	// deal with the - case
```

Comment:
Implement the case where a section is reserved and as a label of the form `433.100-433.110`